Move e2e to api tests and unify test command (#14733)
strapi/strapi: CONTRIBUTING.md

- `yarn test:front` runs front-end related tests.
- `yarn test:front:watch` runs an interactive test watcher for the front-end.
- `yarn test:unit` runs the back-end unit tests.
- `yarn test:api` runs the api integration tests.
- `yarn test:generate-app` generates a test application.
- `yarn test:start-app` starts the test application.

---

## Running the API Integration tests

The API integration tests require a Strapi app to be able to run. You can generate a "test app" using `yarn test:generate-app <database>`:

```bash
$ yarn test:generate-app sqlite
$ yarn test:generate-app postgres
$ yarn test:generate-app mysql
```

> A new app is required every time you run the API integration tests otherwise, the test suite will fail. A command is available to make this process easier: `yarn test:api`.

Reviewer: ```suggestion
A new app is required every time you run the API integration tests, otherwise the test suite will fail. A command is available to make this process easier: `yarn test:api`.
```

> This command runs tests using jest behing the scenes. Options for jest can be passed in the command. (e.g. to update snapshots `yarn test:api -u`).

Reviewer: ```suggestion
This command runs tests using jest behind the scenes. Options for jest can be passed to the command. (e.g. to update snapshots `yarn test:api -u`).
```

### Changing the database

> By default the script test commmand creates an app that uses `sqlite`. But you can run the test suites using different databases:

Reviewer: ```suggestion
By default the script run by `test:api` creates an app that uses `sqlite`. But you can run the test suites using different databases:
```
